lib_eio/process.mli

```diff
@@ -0,0 +1,39 @@
+type status = Exited of int | Signaled of int | Stopped of int
+
+val pp_status : Format.formatter -> status -> unit
+
+class virtual t : object
+    method virtual pid : int
+    method virtual status : status
+    method virtual signal : int -> unit
+end
+
+val pid : #t -> int
+(** The process ID *)
+
+val status : #t -> status
+(** Checks the status of the subprocess, this will block waiting for the subprocess
+    to terminate or be stopped by a signal. *)
+
+val signal : #t -> int -> unit
+(** [signal t i] sends the signal [i] to process [t]. *)
+
+class virtual mgr : object
+    method virtual spawn : 'a 'b 'c.
+        sw:Switch.t ->
+        ?cwd:Fs.dir Path.t ->
+        stdin:(#Flow.source as 'a) ->
+        stdout:(#Flow.sink as 'b) ->
+        stderr:(#Flow.sink as 'c) ->
+        string ->
+        string list ->
+        t
+end
+(** A process manager capable of spawning new processes. *)
+
+val spawn : sw:Switch.t -> #mgr -> ?cwd:Fs.dir Path.t -> stdin:#Flow.source -> stdout:#Flow.sink -> stderr:#Flow.sink -> string -> string list -> t
+(** [spawn ~sw mgr ?cwd ~stdin ~stdout ~stderr cmd args] creates a new subprocess that is connected to the
+    switch [sw]. A process will be stopped when the switch is released.
+    
+    You must provide a standard input and outputs that are backed by file descriptors and
```

Comment:
```suggestion
    [stdin], [stderr] and [stdout] must be backed by file descriptors.
```

Reply:
Thanks, hopefully with the pipe changes I added in https://github.com/ocaml-multicore/eio/pull/473/commits/d553598aeebf293c7ba8f884a17dc32eaeaafe2e this can be completely removed as we check if there's an FD and if not we pipe into the sources and sinks (that code could do with a review I think)